vowpalwabbit/interactions.cc

```diff
@@ -168,6 +210,27 @@ inline bool must_be_left_sorted(const std::vector<namespace_index>& oi)
   return false;  // 'aaa' or 'abc'
 }
 
+std::vector<std::vector<namespace_index>> expand_quadratics_wildcard_interactions(
+    bool leave_duplicate_interactions, const std::set<namespace_index>& new_example_indices)
+{
+  std::set<std::vector<namespace_index>> interactions;
+
+  for (auto it = new_example_indices.begin(); it != new_example_indices.end(); ++it)
+  {
+    auto idx1 = *it;
+    interactions.insert({idx1, idx1});
+
+    for (auto jt = it; jt != new_example_indices.end(); ++jt)
+    {
+      auto idx2 = *jt;
+      interactions.insert({idx1, idx2});
+      interactions.insert({idx2, idx2});
+      if (leave_duplicate_interactions) { interactions.insert({idx2, idx1}); }
+    }
+  }
+  return std::vector<std::vector<namespace_index>>(interactions.begin(), interactions.end());
```

Comment:
is there a way to move all these vectors instead of copying them?

Reply:
Replied in other comment